bin/GoogleAdsCleanupServices.php

```diff
@@ -400,10 +425,26 @@ protected function remove_directory( string $directory ) {
 		$directory = $this->file_path( $directory );
 		if ( ! is_dir( $directory ) ) {
 			$this->warning( sprintf( 'Directory does not exist: %s', $directory ) );
+
 			return;
 		}
 
-		rmdir( $directory ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_rmdir
+		if ( $this->is_empty_directory( $directory ) ) {
```

Comment:
Avoids some errors trying to delete not empty directories.

Reply:
> in which cases does it try to delete a not empty directory?

Maybe it was because when I did this change it was without the changes related to the old structure. (So some enum classes were not deleted from Enum folder) 

> what files are remaining there that we aren't removing? Wouldn't it be better to remove those files as well if needed?

In my point of view not deleting a directory with some not deleted files is less risky (just a matter of space) than deleting them even when they were not deleted by the script (potential functionality class not found problem).

Since you mentioned is not harmful I will keep it just in case. But if is not necessary we can remove it as well.